ns_pipelines/base/base.py

```diff
@@ -124,11 +224,12 @@ def write_pipeline_info(self, hash_outdir: Path):
         }
         FileManager.write_json(hash_outdir / "pipeline_info.json", pipeline_info)
```

Comment:
Write out the output schema for the pipeline here.

Reply:
and a couple lines above ` arg: getattr(self, arg) for arg in inspect.signature(self.__init__).parameters.keys()`, make sure environment variables are not written to file.